### PR TITLE
DEV-1100 — Add details about webhooks with ML backends

### PR DESCRIPTION
**Pull request by smoreface**
Automating pipelines with webhooks and refactor the ML docs a bit.

### CODE REVIEW

File: docs/source/guide/ml_create.md [hunk 1]
```diff
@@ -84,13 +85,25 @@ def predict(self, tasks, **kwargs):
     return predictions
 ```
 
+### Support interactive preannotations in your ML backend
+
+If you want to support interactive preannotations in your machine learning backend, refer to [this code example for substring matching](https://github.com/heartexlabs/label-studio-ml-backend/tree/master/label_studio_ml/examples/substring_matching).
```
Comment (niklub) [inline docs/source/guide/ml_create.md:90]: this is just an example

**Comment by smoreface** [comment]
I'm going to add any non-fit() and non-webhook-related feedback to a different ticket so that this information can be documented sooner. Please focus your review on the new content.

**Comment by smoreface** [comment]
@niklub / @KonstantinKorotaev please take another look when you get a chance, thanks!

File: docs/source/guide/ml_create.md [hunk 2]
```diff
@@ -84,41 +85,63 @@ def predict(self, tasks, **kwargs):
...
+
+Refer to the code example for more details about how this might be performed for a NER labeling project. 
+
 ## Train a model with your ML backend 
 
 If you want to train a model, use the training call to update your model based on new annotations. You can perform training as part of an active learning with predictions, or you can create an ML backend that trains or retrains a model based on annotations. You don't need to use this call in your code if you just want to use an ML backend for predictions. 
 
-Write your own code to override the `fit(completions, **kwargs)` method, which takes [JSON-formatted Label Studio annotations](https://labelstud.io/guide/export.html#Raw-JSON-format-of-completed-labeled-tasks) and returns an arbitrary JSON dictionary where information about the created model can be stored.
+Write your own code to override the `fit()` method, which takes [JSON-formatted Label Studio annotations](/export.html#Raw-JSON-format-of-completed-labeled-tasks) and returns an arbitrary JSON dictionary where information about the created model can be stored.
```
Comment (smoreface) [inline docs/source/guide/ml_create.md:106]: is this sentence still true, or is it misleading with the deprecation?

**Pull request merged**